tests/libnet/cloudinit/factory.go

```diff
@@ -0,0 +1,47 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2024 Red Hat, Inc.
+ *
+ */
+
+package cloudinit
+
+import (
+	. "github.com/onsi/gomega"
+)
+
+func CreateNetworkDataWithStaticIPsByMac(nicName, macAddress, ipAddress string) string {
+	networkData, err := NewNetworkData(
+		WithEthernet(nicName,
+			WithAddresses(ipAddress),
+			WithNameserverFromCluster(),
+			WithMatchingMAC(macAddress),
+		),
+	)
+	ExpectWithOffset(1, err).ToNot(HaveOccurred(), "should successfully create static IPs by mac address cloud init network data")
```

Comment:
I have not changed that behavior, only moved the helper.

But it is possible to add `panic` in case of failure. It is not expected to happen if everything is in order with the cluster and test definition.
I do not think it fits this PR though.

Reply:
Done